org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/DidChangeWatchedFilesRegistrationOptions.java

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.lsp4j.jsonrpc.validation.NonNull;
+import org.eclipse.xtext.xbase.lib.Pure;
+import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
+
+/**
+ * Describe options to be used when registered for text document change events.
+ */
+@SuppressWarnings("all")
+public class DidChangeWatchedFilesRegistrationOptions {
```

Comment:
Shouldn't that should be in lsp4j? If so then please open an enhancement request over there

Reply:
See https://github.com/eclipse/lsp4j/issues/167

Reply:
ok